src/cron/arweave/mod.rs

```diff
@@ -132,20 +129,13 @@ where
     HttpClient: crate::http::Client<Request = reqwest::Request, Response = reqwest::Response>,
 {
     fn get_client(&self) -> HttpClient;
+    fn get_arweave_host(&self) -> String;
```

Comment:
Maybe this should be `get_arweave_uri` or `get_arweave_url` and this should return reference to `Uri` or `Url`.

To me `get_arweave_host` is a bit unclear if it means the full url or just the host part of the url. Especially when it returns a string.

Additionally, I would split this Trait into `ArweaveExecutionContext` and `ArweaveBuildContext` since the url is needed only when `Arweave` object is created. Alternative design would be to move `Arweave` object to live inside the context and then the `get_arweave_(host|url|uri)` is not needed here as it would be internals of the context initialization.

Reply:
Ok, the arweave maintains the same uri from start to end. I guess ArweaveBuildContext suits better